controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -140,7 +140,11 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 	r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateConfiguring
 
 	for _, cephResourceSpec := range r.storageConsumer.Status.CephResources {
-		r.cephResourcesByName[cephResourceSpec.Name] = cephResourceSpec
+		if cephResourceSpec.Name == r.cephClientHealthChecker.Name {
+			r.cephResourcesByName[cephResourceSpec.Name] = cephResourceSpec
+		} else {
+			r.cephResourcesByName[cephResourceSpec.Name] = &ocsv1alpha1.CephResourcesSpec{}
```

Comment:
We need to remove extra entries from the `r.storageConsumer.Status.CephResources` slice

Reply:
Why do we need to do any of it? isn't the status re-written with each reconciliation?

Reply:
> Why do we need to do any of it?
Let's say during upgrade or before upgrade somehow any of storageConsumer cr status is failure, we'll not able to proceed to wait for the status to be the ready state.

> isn't the status re-written with each reconciliation?
we are overriding the status of only healthchecker cephClient in storageConsumer but since in 4.10 we had other status too which will not be updated.

Reply:
no, we will update only health check details in the status. all other states are present in 4.10 (cephblockpool,subvolumegroup, and other ceph clients will not be updated or removed); if any other items are not in a ready state, the consumer onboarding will be stuck in progress.